src/spikeinterface/preprocessing/whiten.py

```diff
@@ -223,27 +198,90 @@ def compute_whitening_matrix(
     # type and we estimate a more reasonable eps in the case
     # where the data is on a scale less than 1.
     if eps is None:
-        eps = 1e-8
-    if data.dtype.kind == "f":
         median_data_sqr = np.median(data**2)  # use the square because cov (and hence S) scales as the square
         if median_data_sqr < 1 and median_data_sqr > 0:
             eps = max(1e-16, median_data_sqr * 1e-3)  # use a small fraction of the median of the squared data
+        else:
+            eps = 1e-16
 
     if mode == "global":
-        U, S, Ut = np.linalg.svd(cov, full_matrices=True)
-        W = (U @ np.diag(1 / np.sqrt(S + eps))) @ Ut
+        W = compute_whitening_from_covariance(cov, eps)
+
     elif mode == "local":
         assert radius_um is not None
         n = cov.shape[0]
         distances = get_channel_distances(recording)
-        W = np.zeros((n, n), dtype="float64")
+        W = np.zeros((n, n), dtype="float32")
```

Comment:
@JoeZiminski from failing tests (might be due to numpy 2.0 and related sklearn version), this seems to require a float64. Any reason why not? this will be rather smalle anyways!
```suggestion
        W = np.zeros((n, n), dtype="float64")
```

Reply:
I'll commit and see if tests pass.